### ISSUE TITLE
Conflict with coordinate system and setting aux array

### ISSUE THREAD

**Issue by cr2940** [clawpack/geoclaw#446]
This is my first time raising an issue in github, so pardon me if I bluff here and there. 

I noticed that in `geoclaw_module.f90` , if `coordinate_system = 1`, we cannot have `mcapa > 0`. But I am trying to make an example where I want to simply use meter system but create a column of small cells, for which I need to use `aux` array. 

Also a question is doesn't this mean that in the case of `coordinate_system=1`, we are forced to set the bathymetry via feeding in a topography data file, and cannot use `setaux` to set up bathymetry?

**Comment by rjleveque** [comment]
I think you might be confusing `mcapa` with `maux`, which is the number of aux arrays (also called `num_aux` in setrun.py).  The `mcapa` variable corresponds to `capa_index` in setrun.py and indicates which of the aux variable corresponds to a ["capacity function"](http://www.clawpack.org/wp_algorithms.html#capacity-functions), the ratio of the physical cell area to the computational grid cell area.  In meters the ratio is 1 so `mcapa` isn't used, but on a lat-long grid it is used.  In either case `aux(1,:,:)` is the topography and you could add other aux variables if needed.

**Comment by cr2940** [comment]
Right, I understand that part. But what if I want to create a non-geo example, a contrived example where I want to set up a non-uniform physical grid?

**Comment by mandli** [comment]
It does seem like you might want to have a capacity array but have `coordinate_system == 1`.  The problem is at geoclaw_module.f90:129.
```fortran
            if ((coordinate_system == 1) .and. (mcapa > 0)) then
                print *, 'ERROR in setgeo:  if coordinate_system = 1 then'
                print *, '      mcapa should be zero'
                stop
            endif
```
@cr2940 is for comparison to an h-box formulation but is still grid-aligned.

**Comment by rjleveque** [comment]
So you will probably have to make your own local version of geoclaw_module.f90 that doesn't throw an error and then change the Makefile to point to it.  And also to your own setaux.f90.

**Comment by mandli** [comment]
If we don't remove the error I would favor writing a more informative message as there isn't necessarily a reason that someone could not use a capacity array with that coordinate system.  Maybe a warning instead?

**Comment by cr2940** [comment]
I will close it. As far as I can see I can simply comment out the terminating statement within the geoclaw module with all things being the same.

**Comment by cr2940** [comment]
Sorry I misread your comment @mandli , i thought you meant close the issue, oops
